ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java

```diff
@@ -932,7 +934,9 @@ public void setDbName(String databaseName) {
   }
 
   public List<FieldSchema> getPartitionKeys() {
-    return tTable.getPartitionKeys();
+    return getStorageHandler() != null && getStorageHandler().alwaysUnpartitioned() &&
+        SessionState.getSessionConf().get(Context.operationNameConf, StringUtils.EMPTY).equals(AlterTableType.COMPACT.getName()) ?
```

Comment:
why do you need to check operationType?

Reply:
Removed this change, not needed anymore.